src/backend/pipeline/streamReceiver.c

```diff
@@ -39,6 +44,21 @@ stream_receive(TupleTableSlot *slot, DestReceiver *self)
 		stream->bytes += tuple->heaptup->t_len + HEAPTUPLESIZE;
 	}
 
+	/*
+	 * Writing to streams from a long-running process is presumably a common case,
+	 * so we periodically commit to avoid a long-running transaction. Long-running xacts
+	 * blocks the autovac from freeing up space, because tuples that have long since been
+	 * deleted would technically still be visible from the point of view of a long-running
+	 * xact.
+	 */
+	if (TimestampDifferenceExceeds(GetCurrentTimestamp(), stream->lastcommit, STREAM_INSERT_COMMIT_INTERVAL))
```

Comment:
We should probably make this into a GUC parameter, with the option of disabling it by setting interval to -1. In cases where there are things other than stream insertions also in the transaction, this behavior might not be ideal.